test/reqQueueDepth.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+// external requires
+var assert = require('chai').assert;
+var restify = require('restify');
+var restifyClients = require('restify-clients');
+
+// local files
+var helper = require('./lib/helper');
+var plugins = require('../lib');
+
+// local globals
+var SERVER;
+var CLIENT;
+var PORT;
+
+
+describe('request queue depth', function () {
+
+    beforeEach(function (done) {
+        SERVER = restify.createServer({
+            dtrace: helper.dtrace,
+            log: helper.getLog('server')
+        });
+
+        SERVER.listen(0, '127.0.0.1', function () {
+            PORT = SERVER.address().port;
+            CLIENT = restifyClients.createJsonClient({
+                url: 'http://127.0.0.1:' + PORT,
+                dtrace: helper.dtrace,
+                retry: false
+            });
+
+            done();
+        });
+    });
+
+    afterEach(function (done) {
+        CLIENT.close();
+        SERVER.close(done);
+    });
+
+
+    it('should throw when missing server argument', function (done) {
```

Comment:
I'd add a few more tests that simulate what happens on 404s, or uncaughtExceptions, or timeouts.